rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -285,6 +291,61 @@ def _maybe_enrich_curve_transfers(
             event.counterparty = CPT_CURVE
             event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} from the curve pool {CURVE_Y_DEPOSIT}'  # noqa: E501
             return True
+
+        # enrich deposits made via curve deposits contracts
+        if (
+            event.event_type == HistoryEventType.SPEND and
+            event.event_subtype == HistoryEventSubType.NONE and
+            source_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
+        ):
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = CPT_CURVE
+            event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} in curve pool'
+            return True
+
+        if (
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            to_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
```

Comment:
Hmmmm I would not rely on such input_data checks.

What if these add / remove liquidity method are called as internal transactions?

Reply:
Internal transactions? That can only happen for ETH transactions right?

Reply:
No-no. Example: https://etherscan.io/tx/0xecc7aedd140923d7760476ed84c00318c943fddddef33111f6bcf957268bcee1/advanced

As you can see here https://ethtx.info/mainnet/0xecc7aedd140923d7760476ed84c00318c943fddddef33111f6bcf957268bcee1/ it calls `RemoveLiquidityOne` method underneath 
![image](https://user-images.githubusercontent.com/63492346/223449048-e0bb45b1-4b6c-4675-9b9f-92518e7c335a.png)